pkg/sql/opt/memo/statistics_builder_test.go

```diff
@@ -99,7 +99,7 @@ func TestGetStatsFromConstraint(t *testing.T) {
 		t.Helper()
 
 		var cols opt.ColSet
-		for i := 0; i < tab.ColumnCount(); i++ {
+		for i := 0; i < tab.AllColumnCount(); i++ {
```

Comment:
I wonder how mutation cols play with stats. A good example of a case where using `ColumnCount` can cause us to not consider a case that is potentially interesting to think about, or at least document.